lib/facter/facts/linux/hypervisors/hyper_v.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Facts
+  module Linux
+    module Hypervisors
+      class HyperV
+        FACT_NAME = 'hypervisors.hyperv'
+
+        def call_the_resolver
+          fact_value = check_hyper_v
+          Facter::ResolvedFact.new(FACT_NAME, fact_value)
+        end
+
+        def check_hyper_v
+          manufacturer = Facter::Resolvers::Linux::DmiBios.resolve(:sys_vendor)
+          product_name = Facter::Resolvers::Linux::DmiBios.resolve(:product_name)
+
+          return {} if manufacturer =~ /Microsoft/ && product_name == 'Virtual Machine'
```

Comment:
should be || instead of && ( https://github.com/puppetlabs/libwhereami/blob/c9666110a1bdec1c7034e4a607710c704603cbf9/lib/src/detectors/hyperv_detector.cc#L12 )

Reply:
Fixed